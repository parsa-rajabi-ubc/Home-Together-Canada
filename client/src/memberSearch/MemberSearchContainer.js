/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.1.17
 *
 * @Description: Member Search Container;
 *
 */

import {React, useState} from 'react';
import ProfileCard from "./ProfileCard";
import MockProfileCardData from "../mockData/MockProfileCardData";
import {USER_TYPES} from "../common/constants/users";
import InvalidUser from "./InvalidUser";
import PropTypes from "prop-types";
import {compose} from "redux";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import BurgerMenu from "../common/listings/BurgerMenu";
import {RemoveScroll} from 'react-remove-scroll';

const MemberSearchContainer = (props) => {
    const [menuOpenState, setMenuOpenState] = useState(false)

    const toggleMenu = () => {
        setMenuOpenState(!menuOpenState);
    }

    const handleMenuOnClose = () => {
        setMenuOpenState(false);
    }

    const {
        accountType,
        authenticated
    } = props;

    // displays 10 mock data
    function profileCards() {
        let profiles = [];
        for (let i = 0; i < 10; i++) {
            profiles.push(<ProfileCard key={i}
                                       username={MockProfileCardData[i].username}
                                       age={MockProfileCardData[i].age}
                                       familyStatus={MockProfileCardData[i].status}
                                       minBudget={MockProfileCardData[i].minRent}
                                       maxBudget={MockProfileCardData[i].maxRent}
                                       pet={MockProfileCardData[i].pet}
                                       smoke={MockProfileCardData[i].smoke}
                                       religion={MockProfileCardData[i].religion}
                                       diet={MockProfileCardData[i].diet}
            />);
        }
        return profiles;
    }

    return (
        <div>
            {/* Checking to Ensure User is Authenticated and is a Member to view this page*/}
            {(!authenticated || accountType !== USER_TYPES.MEMBER) ?
                <InvalidUser/>
                :
                <div className={"flex flex-nowrap"}>
                    {menuOpenState && <RemoveScroll/>}
                    <BurgerMenu isOpen={menuOpenState} onClose={handleMenuOnClose}
                                content={
                                    <div>
                                        <label className={"page-title text-lg text-gray-300 mt-5"}>Modify Your Search
                                            Criteria</label>
                                        <button className={"btn btn-green float-right w-1/5 py-2 px-4 text-base "}
                                                onClick={toggleMenu}>
                                            Done
                                        </button>
                                    </div>
                                }/>
                    {/*Results*/}
                    <div className={"w-1/3 "}>
                        <button className={"btn btn-green w-1/5 py-2 ml-6 mb-6 text-base my-10"}
                                onClick={toggleMenu}>Filter
                        </button>
                        {profileCards()}
                    </div>

                    {/*Map*/}
                    <div className={"flex-1"}>
                        <p>
                            Google Maps Placeholder
                        </p>
                    </div>
                </div>
            }
        </div>
    )
}

const mapStateToProps = (state) => ({
    accountType: state.userPrivileges.accountType,
    authenticated: state.userPrivileges.authenticated
});

MemberSearchContainer.propTypes = {
    authenticated: PropTypes.bool.isRequired,
    accountType: PropTypes.string
}

export default compose(
    withRouter,
    connect(mapStateToProps, null)
)(MemberSearchContainer);
