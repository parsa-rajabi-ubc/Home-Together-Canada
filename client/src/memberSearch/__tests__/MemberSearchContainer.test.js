/**
 * @Author:     Jeff Hatton/Parsa
 * @Created:    2021.1.24
 *
 * @Description: Snapshot test for member search should render properly depending on what is passed as props;
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import MemberSearchContainer from "../MemberSearchContainer";
import {BrowserRouter, Link} from "react-router-dom";

jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));

jest.mock("react-tooltip/node_modules/uuid", () => ({
            v4: () => "00000000-0000-0000-0000-000000000000"}
    )
);

describe('MemberSearchContainer', () => {
    describe('Container test', () => {
        it('should match snapshot test', () => {
            // given
            const props = {
                accountType: null,
                authenticated: false
            };
            //when
            const component = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Invalid user component login button', () => {
        it('should render when the user is not authenticated', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
    });
    describe('Invalid user component sign up button', () => {
        it('should render when the user is not authenticated', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Create an account');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
    });
    describe('Invalid user component login button', () => {
        it('should render when the user is not member', () => {
            // given
            const props = {
                accountType: null,
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
    });
    describe('Invalid user component sign up button', () => {
        it('should render when the user is not member', () => {
            // given
            const props = {
                accountType: null,
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Create an account');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
    });

    describe('Invalid user component login button', () => {
        it('should not render when the user is not authenticated', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(connectWithMembersButton).toBeUndefined();
        });
    });
    describe('Invalid user component sign up button', () => {
        it('should not render when the user is not authenticated', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Sign Up');

            // then
            expect(connectWithMembersButton).toBeUndefined();
        });
    });
    describe('Invalid user component login button', () => {
        it('should not render when the user is not member', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(connectWithMembersButton).toBeUndefined();
        });
    });
    describe('Invalid user component sign up button', () => {
        it('should not render when the user is not member', () => {
            // given
            const props = {
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><MemberSearchContainer {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Sign Up');

            // then
            expect(connectWithMembersButton).toBeUndefined();
        });
    });
});