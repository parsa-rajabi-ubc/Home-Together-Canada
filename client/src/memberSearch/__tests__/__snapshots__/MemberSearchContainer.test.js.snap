// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberSearchContainer Container test should match snapshot test 1`] = `
<div>
  <div />
  <div />
</div>
`;
